index.js

```diff
@@ -141,44 +193,63 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+let ticket = 0
+function takeATicketNumber(line){
+  ticket ++
+  line.push(ticket)
+  return (`Welcome. You are ticket number ${ticket}`)
+}
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log("%c----------", "color: red") 
+console.log(nowServing(line))
+// => `Currently serving 2.`
+
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
+
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
 
-// const newLine = []
-// const takeATicketNumberFunc = ticketNumberGeneratorFunc()
-// console.log(takeATicketNumberFunc(newLine))
-// // => `Welcome. You are ticket number 1`
+function ticketNumberGeneratorFunc(){
+  let ticket = 0
+  function takeATicketNumber(line){
+    ticket ++
+    line.push(ticket)
+    return (`Welcome. You are ticket number ${ticket}`)
+  }
+  return takeATicketNumber
+}
```

Comment:
Excellent! Only small thing - there's no need for the parentheses in the return on line 231. The same is true for other return statements in previous questions. This would work just fine:

    return `Welcome. You are ticket number ${ticketNumber}.`